lib/practice_exercises.rb

```diff
@@ -1,13 +1,58 @@
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 
-# Time Complexity: ?
-# Space Complexity: ?
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
-end
 
-# Time Complexity: ?
-# Space Complexity: ?
+  # handle empty case
+  if string.length == 0
+    return true
+  end
+
+  # format string: downcase and remove anything that's not an alphabetical charachter
+  formatted_string = string.downcase.gsub(/[^A-Za-z]/, '') # yay regex!
+
+  # make copy 
+  formatted_string_copy = formatted_string.clone # allocates new memory 
+
+  # perform reverse algorithim
+  i = 0
+  j = formatted_string_copy.length - 1
+  while i < j
+    formatted_string_copy[i], formatted_string_copy[j] = formatted_string_copy[j], formatted_string_copy[i] # swap values using ruby syntax
+    i += 1
+    j -= 1
+  end 
+
+  # compare formatted strings 
+  if formatted_string == formatted_string_copy
+    return true
+  else 
+    return false
+  end 
+end 
+
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def longest_prefix(strings)
```

Comment:
As  you noted this isn't working.  Take a look at my suggestion below for a potential solution.  Let me know if you have questions.